### PR TITLE
Update dev container and documentations

### PR DESCRIPTION
**Pull request by nguyenhuy0905**
- **What**: delete 2 unnecessary docker-compose and one unnecessary Dockerfile, change dev container to use a Fedora container with CMake instead, and update docs on dev environment. 
- **Why**:
- Containers:
  - The two compose files defined at root directory is very much outdated. It is still using `make` to build the project, while I have already moved to CMake.
  - With CMake being cross-platform by nature (and also has a test system built in), it is more convenient to just use CMake instead.
  - Although some tools are Linux-only (such as, Valgrind), one can simply test the application with valgrind inside the dev container.
- Documentation:
  - Add more details for setting up the (native) development environments on both operating systems (Windows and Unix-like systems).
  - I have done some further testing with the tools on Windows, so I also deleted the disclaimers about Windows not experimented with.
- Note:
  - If you are using NeoVim, I (currently) assume you already know enough to set up the environment yourself. But, that part of the documentation will be updated if requested :smile:.
  - I don't have much experience on Vim/Emacs. If you use that, you're pretty much on your own, for now. Hopefully someone who knows these tools can help with the documentation.
---
Pull request by: Huy Nguyen

### CODE REVIEW

File: .devcontainer/devcontainer.json
Comment (nguyenhuy0905) [inline .devcontainer/devcontainer.json:1]: I could leave it as a Docker container, but a Fedora container has the edge of having newer packages.
Also, I am assuming that your terminal is a 256-color terminal here.
Lastly, I also mount your (possible) SSH keys and the work directory. This makes it possible to access your SSH keys as needed, and retains the changes you make to the codebase.

File: .github/workflows/cmake-clang.yml
Comment (nguyenhuy0905) [inline .github/workflows/cmake-clang.yml:1]: This simply runs this workflow on ALL branches. Same with the other workflow(s).

File: Dockerfile
Comment (nguyenhuy0905) [inline Dockerfile:1]: This is the Dockerfile at the root of this project. As the other 2 docker-composes are not anymore used, this is also, gone.

File: .vale.ini
Comment (nguyenhuy0905) [inline .vale.ini:1]: Configure Vale to only lint Markdown files. Vale uses the ~/vale path by default.

File: README.md
Comment (nguyenhuy0905) [inline README.md:1]: The folder was changed into `docs` a while ago

File: src/entry.c
Comment (nguyenhuy0905) [inline src/entry.c:1]: Nothing happened here, but somehow the file got here.

**Review by nguyenhuy0905 — COMMENTED** [review]
Ah yes, nothing better than reviewing my own "masterpieces"

**Review by BAOELIETRAN — APPROVED** [review]
Amazing good job 👍

**Pull request merged**